core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -307,6 +307,10 @@ class GroupCoordinator(val brokerId: Int,
         }
       }
     }
+
+    // store the group metadata without holding the group lock to avoid the potential
+    // for deadlock when the callback is invoked
+    delayedGroupStore.map(groupManager.store(_))
```

Comment:
Also, are we using the value returned by `groupManager.store`? If not, then `foreach` is more appropriate than `map` (it's a bit clearer and it doesn't allocate a new Option).